TestHelpers.Tests/MockFileSetAttributesTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System.Collections.Generic;
+using NUnit.Framework;
+
+namespace System.IO.Abstractions.TestingHelpers.Tests
+{
+    using XFS = MockUnixSupport;
+
+    [TestFixture]
+    public class MockFileSetAttributesTests
+    {
+
+        [Test]
+        public void MockFile_SetAttributes_ShouldReplaceExistingFlagsIfExistingFile()
+        {
+            var filedata = new MockFileData("test")
+            {
+                Attributes = FileAttributes.Normal
+            };
+            var fileSystem = new MockFileSystem(new Dictionary<string, MockFileData>
+            {
+                {XFS.Path(@"c:\something\demo.txt"), filedata},
+            });
+
+            fileSystem.File.SetAttributes(XFS.Path(@"c:\something\demo.txt"), FileAttributes.Hidden);
+            var attributes = fileSystem.File.GetAttributes(XFS.Path(@"c:\something\demo.txt"));
+            Assert.That(attributes, Is.EqualTo(FileAttributes.Hidden));
+        }
+
+        [Test]
+        public void MockFile_SetAttributes_ShouldReplaceExistingFlagsIfExistingDirectory()
+        {
+            var fileSystem = new MockFileSystem();
+            var path = XFS.Path(@"c:\something");
+            fileSystem.AddDirectory(path);
+            var existingAttributes = fileSystem.File.GetAttributes(path);
+
+            fileSystem.File.SetAttributes(path, existingAttributes | FileAttributes.Hidden);
```

Comment:
The file case makes perfect sense to me. It starts out Normal, and we set the attribute to Hidden. Thus the assertion should be to make sure that the net result is that the file is Hidden (replaced the attribute).

I'm not quite certain why the directory test is pulling the existing attributes, and the File test did not?

Reply:
The explanation is that I wrote the directory ones, and the file one was simply moved from the `MockFileTests.cs` file and into `MockFileSetAttributesTests.cs` to live along side my new tests.

But I think I agree with you. The file and directory tests should be consistent. And the MSDN article on `SetAttributes()` shows you should always add or remove from the existing set of attributes:

https://msdn.microsoft.com/en-us/library/system.io.file.setattributes(v=vs.110).aspx

So my next commit will update the file test with these changes.

Reply:
Ok. Let me grab your branch real quick and mess around with it in my IDE. I still have a couple concerns about how these tests are laid out, but I could be taking crazy pills.

Reply:
Ok, just went through the debugger and confirmed. So, I think how you had it first was correct, given functionality, test name, etc.

Right now you're saying "ShouldREPLACEExistingFlags...", but really what we're doing is ADDing more flags. So the end result is a file that is both "Normal" and "Hidden".

I think that SetAttributes() can be used sort of like a pseudo-RemoveAttribute. i.e. if you have a file thats Hidden -- to remove the hidden and make it normal, I'd think SetAttributes(FileAttributes.Normal) would work. And that would be "replacing".

If you want to preserve all/some of your attributes, then it probably makes sense to selectively remove and then add onto. But if you want to remove all but "Normal" or whatever, I don't think you need to -always- get the current attributes.

Hopefully that wasnt too much of a ramble. The TLDR is that I think it's possible we have two test case scenarios here "replace" and "add additional".

Reply:
I see your point on the file test.

But for directories, they always need to have the `FileAttributes.Directory` flag. So a full replacement isn't really possible.

How about initializing the directory's attributes to something in addition to the `FileAttributes.Directory` flag. Then we can replace it with something else.

Like this:

```c#
[Test]
public void MockFile_SetAttributes_ShouldReplaceExistingFlagsIfExistingDirectory()
{
    var fileSystem = new MockFileSystem();
    var path = XFS.Path(@"c:\something");
    fileSystem.AddDirectory(path);
    var directoryInfo = fileSystem.DirectoryInfo.FromDirectoryName(path);
    directoryInfo.Attributes = FileAttributes.Directory | FileAttributes.Normal;

    fileSystem.File.SetAttributes(path, FileAttributes.Directory | FileAttributes.Hidden);

    var attributes = fileSystem.File.GetAttributes(path);
    Assert.That(attributes, Is.EqualTo(FileAttributes.Directory | FileAttributes.Hidden));
}
```

Reply:
Honestly, the more I think about it and familiarize myself with everything. I don't really see anything wrong with what you're tests are testing. 

I just think the naming is less than ideal and may be a source of confusion (at least for me). Maybe we could just change them to something along the lines of MockFile_SetAttributes_ShouldSetAttributesOnDirectory and MockFile_SetAttributes_ShouldSetAttributesOnFile. Simple and to the point. No worry about what does "replace" mean or are we "adding" more attributes.

I would also drop the..
```
Attributes = FileAttributes.Normal
```
initialization on the File test since that should be set by System.IO.Abstractions for you.

What do you think about that?

Reply:
Sorry, I'll hopefully have a chance to take a look at this again tomorrow.

Life, amiright 😞

Reply:
No worries, we've made you wait years, I'm sure we can wait a little bit. No sweat! 👍